src/IndexRegistry.sol

```diff
@@ -117,121 +126,170 @@ contract IndexRegistry is IndexRegistryStorage {
     *******************************************************************************/
 
     /**
-     * @notice updates the total numbers of operator in `quorumNumber` to `numOperators`
-     * @param quorumNumber is the number of the quorum to update
-     * @param numOperators is the number of operators in the quorum
+     * @notice Increases the historical operator count by 1 and returns the new count
      */
-    function _updateTotalOperatorHistory(uint8 quorumNumber, uint32 numOperators) internal {
-        _totalOperatorsHistory[quorumNumber].push(QuorumUpdate({
-            numOperators: numOperators,
-            fromBlockNumber: uint32(block.number)
-        }));
+    function _increaseOperatorCount(uint8 quorumNumber) internal returns (uint32) {
```

Comment:
feels like maybe a bit many / lot of layering of internal functions?
e.g. `_increaseOperatorCount` is only used in `registerOperator` and `_decreaseOperatorCount` is only used in `deregisterOperator`
not feeling super strongly opinionated about this, but I think this kind of structure can make it a little harder to follow quite what's happening in the logic

Reply:
good call out, thanks. i was feeling similarly - but i think the last implementation was worse - since the layering was there, but a lot of helper functions were similarly-named and served similar functions, or called each other, etc.

i'm gonna keep this as is while i merge some of these PRs, but we should revisit this again and see if it's worth further changes.